test/common/stats/stats_matcher_test.cc

```diff
@@ -0,0 +1,249 @@
+#include "envoy/config/metrics/v2/stats.pb.h"
+
+#include "common/stats/stats_matcher_impl.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+using testing::IsFalse;
+using testing::IsTrue;
+
+namespace Envoy {
+namespace Stats {
+
+class StatsMatcherTest : public testing::Test {
+public:
+  StatsMatcherTest() {}
+  void SetUp() override { stats_matcher_impl_ = std::make_unique<StatsMatcherImpl>(stats_config_); }
+  void TearDown() override {
```

Comment:
I think this should be renamed and explicitly invoked in each test so that it's obvious where the actual testing happens.

For example, make a `void expectAccepted(std::initializer_list<std::string>)` and `void expectRejected(std::initializer_list<std::string>` and then you could replace assignments to expected_to_pass_ or expected_to_fail_ with `checkAccepted({"foo", "bar"})`.

I think that's roughly the same number of lines of test, but easier to understand.

Reply:
I like this idea a lot. Implemented in aa33aaba3.